File: backend-nestjs/src/modules/user-management/application/commands/create-user.command.ts
import { UserRole } from '../../domain/value-objects/user-role';

export class CreateUserCommand {
  constructor(
    public readonly email: string,
    public readonly username: string,
    public readonly password: string,
    public readonly firstName: string,
    public readonly lastName: string,
    public readonly role: UserRole,
    public readonly departmentId?: string,
  ) {}
}

File: backend-dotnet/src/Core/App.Template.Application/Features/UserManagement/Commands/CreateUser/CreateUserCommand.cs
using AppTemplate.Application.DTOs;
using MediatR;

namespace AppTemplate.Application.Features.UserManagement.Commands.CreateUser;

public record CreateUserCommand : IRequest<UserDto>
{
    public string Username { get; init; } = string.Empty;
    public string Email { get; init; } = string.Empty;
    public string Password { get; init; } = string.Empty;
    public string? Name { get; init; }
    public string? Role { get; init; }
    public long? DepartmentId { get; init; }
}
